### PR TITLE
Http response parser

### PR DESCRIPTION
**Pull request by adiraju13**
@RyanLPeterman checkout this code for the http response parser that I wrote. Will create a PR to the other groups repo if you think its gucci.

### CODE REVIEW

File: response_parser.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	std::string r_code = find_response_code(response);
+	if (r_code != "" && is_a_number(r_code))
+		response_code = atoi(r_code.c_str());
+	else
+		return -1;
+
+	//if c_length isnt found, set c_length to equal 0;
+	std::string c_length = find_header_value("Content-Length: ", response);
+	if (c_length != "")
```
Comment (RyanLPeterman) [inline response_parser.cpp]: why do you set content_length to =o if c_length is non empty? seems counterintuitive

File: response_parser.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+std::string ResponseParser::find_response_code(std::string response){
+	int first_space = response.find(" ");
+  	std::string r_code = response.substr(first_space + 1, response.find(" ", first_space + 1) - first_space);
+  	r_code.erase(remove_if(r_code.begin(), r_code.end(), ::isspace), r_code.end());
+  	return r_code;
+}
+
+std::string ResponseParser::find_header_value(std::string header_name, std::string response){
+  	std::string content_length_constant = header_name;
+  	size_t pos = response.find(content_length_constant);
```
Comment (RyanLPeterman) [inline response_parser.cpp]: why not just use header_name directly here?

File: response_parser.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  	if (pos == std::string::npos)
+    	return "";
```
Comment (RyanLPeterman) [inline response_parser.cpp]: indentation fix

File: response_parser.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    std::string header_value = "";
+    //start from the end of the header constant part and gather the value
+  	for (size_t i=pos + content_length_constant.size(); response[i] != '\r'; i++) {
```
Comment (RyanLPeterman) [inline response_parser.cpp]: indentation fix

File: response_parser.cpp (hunk 1 above)
Comment (adiraju13) [inline response_parser.cpp]: shoot this was supposed to be "=="

File: response_parser.cpp (hunk 2 above)
Comment (adiraju13) [inline response_parser.cpp]: will do good catch

File: response_parser.cpp (hunk 4 above)
Comment (adiraju13) [inline response_parser.cpp]: This indentation stuff isnt happening to me locally, so idk what to do

**Pull request merged**